state/src/impls/shard_level.rs

```diff
@@ -1057,6 +1057,51 @@ mod tests {
         ]);
     }
 
+    #[test]
+    fn mint_and_transfer_not_allowed() {
```

Comment:
Would you add a test for a successful case also?

Reply:
tc added for the case of transfer_allowed